composer/cli/launcher.py

```diff
@@ -80,6 +80,13 @@ def _get_parser():
         help=('If set, run the training script as a command (i.e. without `python`). '
               'Cannot be used in conjunction with `module_mode`.'),
     )
+    parser.add_argument('-k',
+                        '--kill_python_zombies',
+                        action='store_true',
+                        help=('If set, at the end of composer, it will run ``kill python`` '
+                              'which will kill all python processes. This can help in the case '
+                              'of NCCL timeouts where child processes become zombies and are '
+                              'not cleaned up.'))
```

Comment:
Let's discuss offline... is it possible to find all children processes and kill only those? then we can turn on by default

Reply:
When I forced a NCCL timeout (and had extra child processes), this was the output:

```
ERROR:composer.cli.launcher:Global rank 0 (PID 66549) exited with code -15
before child processes
after child processes
```

if we run this:

```
if args.kill_python_zombies:
    current_process = psutil.Process()
    children = current_process.children(recursive=True)
    print ("before child processes")
    for child in children:
        print('Child pid is {}'.format(child.pid))
    print ("after child processes")
```

so it feels as if we're unable to actually get children when there is a NCCL timeout. At least in this naive manner.